### PR TITLE
added listener for bread delete for menu and permission removal

### PR DESCRIPTION
**Pull request by ntuple**
**Delete** the **Menu** and **Permissions** associated with a **Bread** if the Bread is deleted.

### CODE REVIEW

File: src/Listeners/DeleteBreadMenuItem.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    /**
+     * Delete a MenuItem for a given BREAD.
+     *
+     * @param BreadDeleted $bread
+     *
+     * @return void
+     */
+    public function handle(BreadDeleted $bread)
+    {
+        if (config('voyager.bread.add_menu_item') && file_exists(base_path('routes/web.php'))) {
```
Comment (fletch3555) [inline src/Listeners/DeleteBreadMenuItem.php]: What's the purpose of checking for the route file?

File: src/Listeners/DeleteBreadPermission.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    /**
+     * Delete Permission for a given BREAD.
+     *
+     * @param BreadDeleted $bread
+     *
+     * @return void
+     */
+    public function handle(BreadDeleted $bread)
+    {
+        if (config('voyager.bread.add_permission') && file_exists(base_path('routes/web.php'))) {
```
Comment (fletch3555) [inline src/Listeners/DeleteBreadPermission.php]: Again with the route file

File: src/Listeners/DeleteBreadPermission.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            // Permission::generateFor(snake_case($bread->dataType->slug));
+            $role = Role::where('name', config('voyager.bread.default_role'))->firstOrFail();
```
Comment (fletch3555) [inline src/Listeners/DeleteBreadPermission.php]: None of this is necessary.  I'd say it's safe to assume the database is set up correctly with ONDELETE, and will delete the associated pivot table rows when the permissions themselves are deleted.

File: src/Listeners/DeleteBreadPermission.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+            // Get permission for added table
+            $permissions = Permission::where(['table_name' => $bread->dataType->name])->get()->pluck('id')->all();
+
+            // Assign permission to admin
+            $role->permissions()->detach($permissions);
```
Comment (fletch3555) [inline src/Listeners/DeleteBreadPermission.php]: You're not actually deleting the permissions anywhere, just removing them from the default user.  Instead, they should be deleted and removed from all roles (ideally automatically per my comment above, or manually if soft-delete is used)

**Review by fletch3555 — CHANGES_REQUESTED** [review]

File: src/Listeners/DeleteBreadMenuItem.php [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+     * Delete a MenuItem for a given BREAD.
+     *
+     * @param BreadDeleted $bread
+     *
+     * @return void
+     */
+    public function handle(BreadDeleted $bread)
+    {
+        if (config('voyager.bread.add_menu_item')) {
+            $menuItem = MenuItem::where(['title' => $bread->dataType->display_name_plural]);
```
Comment (emptynick) [inline src/Listeners/DeleteBreadMenuItem.php]: I think its more unique/safe to select the menu-item by the route.
```suggestion
            $menuItem = MenuItem::where('route', 'voyager.'.$bread->dataType->slug.'.index');
```

**Review by emptynick — DISMISSED** [review]

**Review by fletch3555 — APPROVED** [review]

**Pull request merged**